content/ch-gates/oracles.ipynb

```diff
@@ -0,0 +1,289 @@
+    "\n",
+    "In order to understand how an oracles work within a quantum algorithm, we need to be specific about how they are defined. One of the  main forms that oracles take is that of *Boolean oracles*. These are described by the following unitary evolution,\n",
+    "\n",
+    "$$\n",
+    "U_f \\left|x , \\bar 0 \\right\\rangle = \\left|x, f(x)\\right\\rangle.\n",
+    "$$\n",
+    "\n",
+    "Here $\\left|x , \\bar 0 \\right\\rangle = \\left|x \\right\\rangle \\otimes \\left|\\bar 0 \\right\\rangle$ is used to represent a multi-qubit state consisting of two registers. The first register is in state $\\left|x\\right\\rangle$, where $x$ is a binary representation of the input to our function. The number of qubits in this register is the number of bits required to represent the inputs.\n",
+    "\n",
+    "The job of the second register is to similarly encode the output. Specifically, the state of this register after applying $U_f$ will be a binary representation of the output $\\left|f(x)\\right\\rangle$, and this register will consist of as many qubits as are required for this. This initial state $\\left|\\bar 0 \\right\\rangle$ for this register represents the state for which all qubits are $\\left|0 \\right\\rangle$. For other initial states, applying $U_f$ will lead to different results. The specific results that arise will depend on how we define the unitary $U_f$.\n",
+    "\n",
+    "Another form of oracle is the *Phase oracle*. In this case, there is no additional register to hold the output $f(x)$. Instead its value determines the phase acquired by a given state\n",
+    "\n",
+    "$$\n",
+    "P_f \\left|x \\right\\rangle = (-1)^{f(x)} \\left|x \\right\\rangle.\n",
+    "$$\n",
+    "\n",
+    "For phase oracles, the output $f(x)$ is typically a simple bit value of $0$ or $1$, corresponding to inducing phaes of $+1$ and $-1$, respectively."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Taking Out the Garbage <a id='garbage'></a>\n",
+    "\n",
+    "The functions evaluated by an oracle are typically those that can be evaluated efficiently on a classical computer. However, the need to implement it as a unitary in one of the forms shown above means that it must instead be implemented using quantum gates. However, this is not quite as simple as just taking the Boolean gates that can implement the classical algorithm, and replacing them with their classical counterparts.\n",
+    "\n",
+    "One issue that we must take care of is that of reversibility. A unitary of the form $U = \\sum_x \\left| f(x) \\right\\rangle \\left\\langle x \\right|$ is only possible if every unique input $x$ results in a unique output $f(x)$, which is not true in general. However, we can force it to e true by simply including a copy of the input in the output.\n",
+    "\n",
```

Comment:
letter 'b' is missing, in the sentence "However, we can force it to e true by simply including a copy of the input in the output."